lib/AST/Type.cpp

```diff
@@ -2888,7 +2892,33 @@ Optional<ProtocolConformanceRef>
 LookUpConformanceInSubstitutionMap::operator()(CanType dependentType,
                                        Type conformingReplacementType,
                                        ProtocolType *conformedProtocol) const {
-  return Subs.lookupConformance(dependentType, conformedProtocol->getDecl());
+  auto conformance =
+      Subs.lookupConformance(dependentType, conformedProtocol->getDecl());
+  if (conformance.hasValue())
+    return conformance;
+
+  if (!conformingReplacementType)
+    return None;
+
+  if (auto nominal =
```

Comment:
It looks like you're working around a problem in lookupConformance() here? Can you add comments or explain in the review